worker/caasapplicationprovisioner/application.go

```diff
@@ -402,7 +406,11 @@ func (a *appWorker) ensureTrust(app caas.Application) error {
 	}
 
 	a.logger.Debugf("updating application %q trust to %v", a.name, desiredTrust)
-	if err := app.Trust(desiredTrust); err != nil {
+	err = app.Trust(desiredTrust)
+	if errors.IsNotFound(err) {
+		return nil
```

Comment:
maybe log a warning?

This is for the case where the app gets removed before we can act on the trust request right? I think in similar cases elsewhere do also ignore the not found, is that right?

Reply:
This could happen when trust watcher fired event before ensureServer gets called. 
So I changed to return `dependency.ErrBounce` to restart the worker to retry